util.go

```diff
@@ -44,3 +45,35 @@ func (genmsg *NFGenMsg) Decode(b []byte) {
 	genmsg.Version = b[1]
 	genmsg.ResourceID = binary.BigEndian.Uint16(b[2:])
 }
+
+// GetFirstIPFromCIDR returns the first IP address from a CIDR.
+func GetFirstIPFromCIDR(cidr string) (*net.IP, error) {
+	_, subnet, err := net.ParseCIDR(cidr)
+	if err != nil {
+		return nil, err
+	}
+
+	mask := binary.BigEndian.Uint32(subnet.Mask)
+	ip := binary.BigEndian.Uint32(subnet.IP)
+
+	// find the final address
+	firstIP := make(net.IP, 4)
+	binary.BigEndian.PutUint32(firstIP, ip&mask)
+
+	return &firstIP, nil
+}
+
+// GetLastIPFromCIDR returns the last IP address from a CIDR.
+func GetLastIPFromCIDR(cidr string) (*net.IP, error) {
+	_, subnet, err := net.ParseCIDR(cidr)
+	if err != nil {
+		return nil, err
+	}
+	mask := binary.BigEndian.Uint32(subnet.Mask)
+	ip := binary.BigEndian.Uint32(subnet.IP)
+	// find the final address
+	lastIP := make(net.IP, 4)
+	binary.BigEndian.PutUint32(lastIP, (ip&mask)|(mask^0xffffffff))
```

Comment:
What does the bitwise operation do? Deserves a comment.

Reply:
This doesn’t seem to be resolved? I don’t see a comment explaining the bitwise operation.

Reply:
I'm sorry, I've commented on another bitwise operation in nftables_test.go. Now I've added the comment here. Let me know if you think it is clear.